Natural_gradient

ujohn33 opened the pull request:
Added an option for training with natural gradient and gradient clipping as hyperparameter (quanitle and threshold types)

StatMixedML commented:
@ujohn33 Thanks for opening the PR.

What's the reason why you closed this again?

ujohn33 commented:
Hi Alexander, 

I have reopened the PR now.
I have added a few more commits since I added and I was not sure if they apply but apparently they do. 

The change was that I sped up the quantile clipping by doing it with only torch avoiding nympy.

Best regards,
Evgeny

> On 13 Jan 2025, at 12:14, Alexander März ***@***.***> wrote:
> 
> 
> @ujohn33 <https://github.com/ujohn33> Thanks for opening the PR.
> 
> What's the reason why you closed this again?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/StatMixedML/LightGBMLSS/pull/45#issuecomment-2586565687>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AMUAW2PNXUKFOWWXPOUSKE32KN7YBAVCNFSM6AAAAABU6ER3HOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOBWGU3DKNRYG4>.
> You are receiving this because you were mentioned.
>

StatMixedML commented:
@ujohn33 Thanks for opening the PR again. Can you please add unit-tests for your changes. Thanks
